src/decommit.rs

```diff
@@ -91,11 +108,21 @@ impl WorldDiff {
         self.decommitted_hashes.insert(decommit.code_key, ());
         Some(world.decommit(decommit.code_key))
     }
+
+    pub(crate) fn unpaid_decommit(
```

Comment:
No reason to mimic the two function split. Just make one function that does the whole decommit for decommit opcode and leave `UnpaidDecommit` alone.